### PR TITLE
Fix broken actions (again)

### PR DESCRIPTION
**Pull request by yardasol**
## Summary of changes

OpenMC [PR 2116](https://github.com/openmc-dev/openmc/pull/2116) added support for MCPL files, and modifies the `gha-install.py` script to require MCPL support. This PR adds a line to SaltProcs `build-openmc.sh` that runs the script in OpenMC to install MCPL.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Required for Merging
- [x] I have read the **CONTRIBUTING** document.
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly. 
- [ ] My change is a source code change
  - [ ] I have added/modified tests to cover my changes
  - [ ] I have explained why my change does not require new/modified tests
- [ ] My change is a user-facing change
  - [ ] I have recorded my changes in the changelog for the upcoming release
- [x] My change is exclusively related to the repository (e.g. GitHub actions workflows, PR/issue templates, etc.)
  - [x] I have verified or justified that my change will work as intended.
- [x] All new and existing tests passed.
  - [x] CI tests pass
  - [x] Local tests pass (including Serpent2 integration tests)
## Associated Issues and PRs

None

## Associated Developers

- Dev: @munkm 

## Checklist for Reviewers

Reviewers should use [this link](https://arfc.github.io/manual/guides/pull_requests) to get to the 
Review Checklist before they begin their review.

### CODE REVIEW

File: .github/workflows/cache-dependencies.yml [hunk 1]
```diff
@@ -76,6 +76,7 @@ jobs:
         with: 
           repository: openmc-dev/openmc
           path: openmc
+          submodules: reursive
```
Comment (abachma2) [inline .github/workflows/cache-dependencies.yml:79]: I am fairly unexperienced with github workflows. What does this line do in the workflow?

File: tools/ci/build-openmc.sh [hunk 2]
```diff
@@ -3,6 +3,7 @@ set -ex
 
 # Build openmc
 cd openmc
+./tools/ci/gha-install-mcpl.sh
```
Comment (abachma2) [inline tools/ci/build-openmc.sh:6]: Is this a hardcoded path? Will there ever be a time when this needs to be modified?

**Review by abachma2 — APPROVED** [review]
Hi @yardasol. The PR looks good and like it will accomplish what you need. I have approved this PR, but I do have a few questions.

File: tools/ci/build-openmc.sh (hunk 2 above)
Comment (yardasol) [reply tools/ci/build-openmc.sh:6]: Only if the corresponding path changes in OpenMC's repo

Comment (abachma2) [reply tools/ci/build-openmc.sh:6]: This is within the OpenMC file structure, not anything the user would be defining?

File: .github/workflows/cache-dependencies.yml (hunk 1 above)
Comment (yardasol) [reply .github/workflows/cache-dependencies.yml:79]: See the [README](https://github.com/actions/checkout#usage) for the ``checkout`` action. We set this setting to ``recursive`` to match the command in the [instructions to install OpenMC from source.](https://docs.openmc.org/en/latest/usersguide/install.html#obtaining-the-source)

Comment (yardasol) [inline .github/workflows/cache-dependencies.yml:79]:
```suggestion
          submodules: recursive
```

File: tools/ci/build-openmc.sh (hunk 2 above)
Comment (yardasol) [reply tools/ci/build-openmc.sh:6]: Correct!

**Comment by abachma2** [comment]
Thanks for answering my questions @yardasol. Merging this PR!

**Pull request merged**